Fix deactivation of gtk-doc

The tarball contains the Makefile for building documentation with gtk-doc,
Unfortunately the AM_CONDITIONAL variable is not the correct one, which
results in an error when running autoreconf.

This patch fixes this issue.

Signed-off-by: Eric Le Bihan <eric.le.bihan.dev@free.fr>
---
Index: systemd-213/docs/gtk-doc.make
===================================================================
--- systemd-213.orig/docs/gtk-doc.make	2014-02-14 09:27:47.000000000 +0100
+++ systemd-213/docs/gtk-doc.make	2014-06-04 15:20:11.976845074 +0200
@@ -267,7 +267,7 @@
 #
 # Require gtk-doc when making dist
 #
-if HAVE_GTK_DOC
+if ENABLE_GTK_DOC
 dist-check-gtkdoc: docs
 else
 dist-check-gtkdoc:
